wrangler-core/src/main/java/io/cdap/wrangler/utils/Json2Schema.java

```diff
@@ -73,62 +73,75 @@ public Schema toSchema(String id, Row row) throws RecordConvertorException {
+      try {
+        return Schema.nullableOf(new SimpleSchemaGenerator().generate(value.getClass()));
+
+      } catch (UnsupportedTypeException e) {
+        throw new RecordConvertorException(
+          String.format("Unable to convert field '%s' to basic type.", name)
+        );
       }
+    }
 
-      if (value instanceof LocalDate) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.LogicalType.DATE));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof BigDecimal) {
+      // TODO CDAP-15361 precision should be derived from the schema of the row.
+      return Schema.nullableOf(Schema.decimalOf(38, ((BigDecimal) value).scale()));
+    }
 
-      if (value instanceof LocalTime) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MICROS));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof LocalDate) {
+      return Schema.nullableOf(Schema.of(Schema.LogicalType.DATE));
+    }
 
-      if (value instanceof ZonedDateTime) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof LocalTime) {
+      return Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MICROS));
+    }
 
-      // TODO - remove all the instaces of java.util.Date once all the directives support LogicalType.
-      if (value instanceof Date || value instanceof java.sql.Date || value instanceof Time
-        || value instanceof Timestamp) {
-        Schema schema = Schema.nullableOf(Schema.of(Schema.Type.LONG));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof ZonedDateTime) {
+      return  Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS));
+    }
 
-      if (value instanceof Map) {
-        Schema schema = Schema.nullableOf (Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING)));
-        fields.add(Schema.Field.of(name, schema));
-      }
+    // TODO - remove all the instaces of java.util.Date once all the directives support LogicalType.
+    if (value instanceof Date || value instanceof java.sql.Date || value instanceof Time
+      || value instanceof Timestamp) {
+      return Schema.nullableOf(Schema.of(Schema.Type.LONG));
+    }
 
-      if (value instanceof JsonElement) {
-        Schema schema = toSchema(name, (JsonElement) value);
-        fields.add(Schema.Field.of(name, schema));
-      }
+    if (value instanceof Map) {
+      return Schema.nullableOf (Schema.mapOf(Schema.of(Schema.Type.STRING), Schema.of(Schema.Type.STRING)));
+    }
 
-      if (value instanceof Schema) {
-        fields.addAll(((Schema) value).getFields());
+    if (value instanceof JsonElement) {
+      return toSchema(name, (JsonElement) value);
+    }
+
+    if (value instanceof List) {
+      List<?> list = (List) value;
+      for (Object listObject : list) {
```

Comment:
if the list is empty, a null schema is returned. There's not really anything you can do about this, but do you know what that looks like in the UI?

Reply:
If schema is null, we do not add that field schema into schema in line 83. Meaning, UI will show type as unknown. Which is same as current behavior. In that case UI shows type as `unknown`.

![image](https://user-images.githubusercontent.com/14131070/57552309-58f35300-7320-11e9-9e79-58ae7030d3be.png)

![image](https://user-images.githubusercontent.com/14131070/57552334-6f99aa00-7320-11e9-89b8-16e9e05d8e84.png)